metricbeat/module/kubernetes/util/enrichers.md

```diff
@@ -0,0 +1,88 @@
+## Kubernetes Metadata enrichment
+
+The metadata enrichment process involves associating contextual information, such as Kubernetes metadata (e.g., labels, annotations, resource names), with metrics and events collected by Elastic Agent and Beats in Kubernetes environments. This process enhances the understanding and analysis of collected data by providing additional context.
+
+### Key Components:
+
+1. **Metricsets/Datasets:**
+   - Metricsets/Datasets are responsible for collecting metrics and events from various sources within Kubernetes, such as kubelet and kube-state-metrics.
```

Comment:
Still dont explain why we have two terms. Users dont understand why this changes in beats vs agents. I think we should make this clear

Reply:
@gizas  This markdown is in beats repo, under kubernetes module. There, only metricsets exist. Personally I would remove the `dataset` as it is confusing. What kind of sentence do you want me to add?

Reply:
Missed that we are under beats. I had in my mind that this could be a more general doc. 
So ok you can remove datasets overall